lib/hdt.js

```diff
@@ -54,20 +54,28 @@ HdtDocumentPrototype.searchTerms = function (options) {
       position = options.position,
       posId = POSITIONS[position],
       prefix = options.prefix || '',
+      subject = options.subject,
       object = options.object;
   if (!(position in POSITIONS))
     return Promise.reject(new Error('Invalid position argument. Expected subject, predicate or object.'));
 
+  if ('subject' in options || 'object' in options) {
+    if (subject === '') return Promise.reject(new Error('Unspecified subject value.'));
+    if (object === '')  return Promise.reject(new Error('Unspecified object value.'));
```

Comment:
When would the above two conditions happen? Who would specify subject and object, but specify them as the empty string?

Reply:
That is in case somehow the function is called with an empty string in the subject or object position:
```JavaScript
searchTerms(subject: '', position: 'predicate')
searchTerms(object: '', position: 'predicate')
searchTerms(subject: '', object: '', position: 'predicate')
```

Reply:
> That is in case somehow the function is called with an empty string

Yes but why would that case happen? And why do we need to throw an exception for that instead of just not returning results (as I expect would happen without this line)?

I don't see why we need to catch the above case while we're not catching:

```
searchTerms({ subject: null, position: 'predicate', object: 1234 })
```

Reply:
Sorry, I now see your point and I agree. Let's make searchTerms return an empty array in case given subject or object is undefined, null or empty string. Is that ok?

Reply:
Pushed changes related to this conversation.